src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -464,19 +518,45 @@ private BlockPos getRandomBuilding()
             return worker.getPosition();
         }
 
-        final Collection<AbstractBuilding> buildingList = worker.getColony().getBuildings().values();
-        final Object[] buildingArray = buildingList.toArray();
+        final Random random = new Random();
+
+        BlockPos pos = null;
+        while(pos == null
+                || world.getBlockState(pos).getMaterial().isLiquid()
+                || !world.getBlockState(pos.down()).getMaterial().isSolid()
+                || (!world.isAirBlock(pos) && !world.isAirBlock(pos.up())))
+        {
+            final Tuple<EnumFacing, EnumFacing> direction = getRandomDirectionTuple(random);
+            pos =
+                    new BlockPos(worker.getPosition())
+                            .offset(direction.getFirst(), random.nextInt(10))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)